sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestBindProperties.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.sqlobject;
+
+import java.util.List;
+
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.mapper.ImmutableTrain;
+import org.jdbi.v3.core.mapper.ImmutablesTest.Train;
+import org.jdbi.v3.core.mapper.reflect.ImmutablesMapperFactory;
+import org.jdbi.v3.core.rule.H2DatabaseRule;
+import org.jdbi.v3.sqlobject.customizer.BindProperties;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestBindProperties {
+    @Rule
+    public H2DatabaseRule dbRule = new H2DatabaseRule().withPlugin(new SqlObjectPlugin());
+
+    private Handle h;
+
+    private Dao dao;
+
+    @Before
+    public void setUp() {
+        h = dbRule.getSharedHandle();
+        h.registerRowMapper(ImmutablesMapperFactory.mapImmutable(Train.class, ImmutableTrain.class, ImmutableTrain::builder));
```

Comment:
Feels like we want annotations to register Immutables mappers, e.g. `@RegisterImmutablesMapper(value = Train.class)` annotation. The impl and builder method name could be default attributes that look for `Immutable*` and `Immutable*.builder()`, respectively.

Reply:
While we're on the subject.. those automatic values feel like good parameter defaults on the immutable and modifiable factory methods themselves.

Reply:
We could name the annotations `@RegisterRowMapper.Immutable` and `@RegisterRowMapper.Modifiable` just to be smart asses :trollface:

Reply:
Yeah it's a bit of a misnomer to mix Immutable and Modifiable together, but that's sort of what the Immurables library hands us with their `@Value.Immutable` and `@Value.Modifiable` annotations. I'm not sure how to advertise the "Modifiable" support without being confusing.

Reply:
The new approach is to have an `ImmutablesPlugin` with `forImmutable` and `forModifiable` static factory methods.  how's that?

Reply:
Instead of making a new annotation, I made a plugin.  Hopefully that's a good solution.